app/src/test/java/io/gnosis/safe/ui/safe/settings/SafeSettingsViewModelTest.kt

```diff
@@ -26,36 +28,41 @@ class SafeSettingsViewModelTest {
 
     private lateinit var safeSettingsViewModel: SafeSettingsViewModel
 
-    @Before
-    fun setup() {
-        safeSettingsViewModel = SafeSettingsViewModel(safeRepository, tracker, appDispatchers)
-    }
 
     @Test
     fun `removeSafe - should remove safe`() = runBlockingTest {
 
         coEvery { safeRepository.getActiveSafe() } returnsMany listOf(SAFE_1, null)
+        coEvery { safeRepository.activeSafeFlow() } returns flow {
+            emit(SAFE_1)
+            emit(null)
+        }
+            .conflate()
+        coEvery { safeRepository.setActiveSafe(any()) } just Runs
         coEvery { safeRepository.getSafes() } returnsMany listOf(SAFES, listOf(SAFE_2))
         coEvery { safeRepository.removeSafe(ACTIVE_SAFE) } just Runs
         coEvery { tracker.setNumSafes(any()) } just Runs
 
+        safeSettingsViewModel = SafeSettingsViewModel(safeRepository, tracker, appDispatchers)
+        val stateObserver = TestLiveDataObserver<BaseStateViewModel.State>()
+        safeSettingsViewModel.state.observeForever(stateObserver)
+
         val safeCount = safeRepository.getSafes().count()
         assert(safeCount == 2)
 
         safeSettingsViewModel.removeSafe()
 
-        safeSettingsViewModel.state.test().assertValueAt(0) {
-            it is SafeSettingsState.SafeRemoved &&
-                    it.viewAction is BaseStateViewModel.ViewAction.NavigateTo
-        }
-
-        safeSettingsViewModel.state.test().assertValueAt(0) {
-            it is SafeSettingsState.SafeSettings &&
-                    it.safe == null
+        with(stateObserver.values()[0]) {
+            val viewAction = this.viewAction
+            assert (
+                viewAction is BaseStateViewModel.ViewAction.UpdateActiveSafe &&
+                        viewAction.newSafe == null
```

Comment:
So we did have a `SafeSettingsState.SafeRemoved`? I would return to that pattern where we have a viewAction that represents `SafeRemoved`